test/testUtils.js

```diff
@@ -462,6 +473,30 @@ module.exports.utils = {
 
         test.equals(utils.getLocaleFromPath('[dir]/strings_[localeUnder].json', "x/y/strings_zh_Hans_CN.json"), "zh-Hans-CN");
 
+        test.done();
+    },
+
+    testGetLocaleFromPathLocaleLower: function(test) {
+        test.expect(1);
+
+        test.equals(utils.getLocaleFromPath('[dir]/strings_[localeLower].json', "x/y/strings_de-de.json"), "de-DE");
+
+        test.done();
+    },
+
+    testGetLocaleFromPathLocaleLowerShort: function(test) {
+        test.expect(1);
+
+        test.equals(utils.getLocaleFromPath('[dir]/strings_[localeLower].json', "x/y/strings_de.json"), "de");
+
+        test.done();
+    },
+
+    testGetLocaleFromPathLocaleLowerLong: function(test) {
+        test.expect(1);
+
+        test.equals(utils.getLocaleFromPath('[dir]/strings_[localeLower].json', "x/y/strings_zh-hans-cn.json"), "zh-hans-CN");
```

Comment:
is `zh-hans-CN` acceptable in this case?
Locale class automatically transforms region to upper case (see L482), but it doesn't work with script.

We could consider making first letter of a script upper cased by default by Locale class.

Reply:
It should be all lower case, exactly how you have it. Some packages, like Django unfortunately, use the non-standard all-lower-case locale specs and rely on the correct ordering of the parts instead of telling the difference by the capitalization. Django doesn't use the script part so the Chinese locale is just "zh-cn", with no "hans" part.

Reply:
I was more concerned about reverse parsing of the locale by loctool which is then used internally.

Notice that `zh-hans-cn` was parsed by loctool as `zh-hans-CN`, not `zh-Hans-CN`. There in loctool code of parsing the locale out of filename (by using a template) last part of the locale (`CN` in this example) is automatically transformed to upper case. (https://github.com/iLib-js/iLib/blob/development/js/lib/Locale.js#L120)

Does it make sense to enforce smth like `ucfirst` for the script part of the locale too, just to have consistency between `SourceLocale <=> ParsedLocale`? As far as I can see here: https://www.unicode.org/iso15924/iso15924-codes.html - all script codes start with an upper-cased letter.